File: survival_analytics.r
############################## problem1 #######################################
# Survival Analysis in R
library(readr)
install.packages('survminer')
install.packages("survival")

library(survminer)
library(survival)


patient_data <- read.csv("E:\\DATA SCIENCE ASSIGNMENT\\Class And Assignment Dataset\\Asss\\Survival Analytics\\Patient.csv",stringsAsFactors = TRUE)

attach(patient_data)
str(patient_data)

#EDA
patient_data <- patient_data[,c(2,3,4)]
str(patient_data)
colnames(patient_data)
# Define variables 
time <- Followup
event <- Eventtype
group <- Scenario  # unemployment insurance can take 2 values 0 or 1 

# Descriptive statistics
summary(time)
table(event)
table(group)

# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time, event) ~ 1)

summary(kmsurvival)

plot(kmsurvival, xlab="Time", ylab="Survival Probability")

ggsurvplot(kmsurvival, data=patient_data, risk.table = TRUE)


# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ group)
summary(kmsurvival1)

plot(kmsurvival1, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=survival_unemployment1, risk.table = TRUE)

############################# problem2 ##########################################
# Survival Analysis in R
library(readr)
install.packages('survminer')
install.packages("survival")

library(survminer)
library(survival)


ECG_data <- readxl::read_excel("E:\\DATA SCIENCE ASSIGNMENT\\Class And Assignment Dataset\\Asss\\Survival Analytics\\ECG_Surv.xlsx")

attach(ECG_data)
str(ECG_data)

#EDA
ECG_data <- ECG_data[,c(1,2,12)]
str(ECG_data)
colnames(ECG_data)
# Define variables 
time <- survival_time_hr
event <- alive
group <- group  # unemployment insurance can take 2 values 0 or 1 

# Descriptive statistics
summary(time)
table(event)
table(group)

# Kaplan-Meier non-parametric analysis
kmsurvival <- survfit(Surv(time, event) ~ 1)

summary(kmsurvival)

plot(kmsurvival, xlab="Time", ylab="Survival Probability")

ggsurvplot(kmsurvival, data=ECG_data, risk.table = TRUE)


# Kaplan-Meier non-parametric analysis by group
kmsurvival1 <- survfit(Surv(time, event) ~ group)
summary(kmsurvival1)

plot(kmsurvival1, xlab="Time", ylab="Survival Probability")
ggsurvplot(kmsurvival1, data=ECG_data, risk.table = TRUE)
############################## END ############################################

File: survival_analytics.py
#################################### problem1 ###################################################
# pip install lifelines
import lifelines

import pandas as pd
# Loading dataset
patient_data = pd.read_csv("E:/DATA SCIENCE ASSIGNMENT/Class And Assignment Dataset/Asss/Survival Analytics/Patient.csv")
patient_data.head()
patient_data.describe()
patient_data.columns
patient_data["Followup"].describe()


#EDA
#Removing PatientID column
patient_data = patient_data.iloc[:,1:4]

#label enconding
from sklearn import preprocessing
label_encoder = preprocessing.LabelEncoder()
patient_data['Scenario'] = label_encoder.fit_transform(patient_data['Scenario'])

# Followup is referring to time 
T = patient_data.Followup

# Importing the KaplanMeierFitter model to fit the survival analysis
from lifelines import KaplanMeierFitter

# Initiating the KaplanMeierFitter model
kmf = KaplanMeierFitter()

# Fitting KaplanMeierFitter model on Time and Events for Eventtype 
kmf.fit(T, event_observed=patient_data.Eventtype)

# Time-line estimations plot 
kmf.plot()

# Over Multiple Scenario 
# For each group, here group is Scenario
patient_data.Scenario.value_counts()

# Applying KaplanMeierFitter model on Time and Events for the Scenario "1"
kmf.fit(T[patient_data.Scenario==0], patient_data.Eventtype[patient_data.Scenario==0], label='1')
ax = kmf.plot()

# Applying KaplanMeierFitter model on Time and Events for the Scenario "0"
kmf.fit(T[patient_data.Scenario==0], patient_data.Eventtype[patient_data.Scenario==0], label='0')
kmf.plot(ax=ax)


############################################ problem2 ###############################################
# pip install lifelines
import lifelines

import pandas as pd
# Loading dataset
ECG_data = pd.read_excel("E:/DATA SCIENCE ASSIGNMENT/Class And Assignment Dataset/Asss/Survival Analytics/ECG_Surv.xlsx")
ECG_data.head()
ECG_data.describe()
ECG_data.columns
ECG_data["survival_time_hr"].describe()

#EDA
#Selecting required column
ECG_data = ECG_data[["survival_time_hr","alive","group"]]

# survival_time_hr is referring to time 
T = ECG_data.survival_time_hr

# Importing the KaplanMeierFitter model to fit the survival analysis
from lifelines import KaplanMeierFitter

# Initiating the KaplanMeierFitter model
kmf = KaplanMeierFitter()

# Fitting KaplanMeierFitter model on Time and Events for alive 
kmf.fit(T, event_observed=ECG_data.alive)

# Time-line estimations plot 
kmf.plot()

# Over Multiple group 
# For each group, here group is group
ECG_data.group.value_counts()

# Applying KaplanMeierFitter model on Time and Events for the group "1"
kmf.fit(T[ECG_data.group==1], ECG_data.alive[ECG_data.group==1], label='1')
ax = kmf.plot()

# Applying KaplanMeierFitter model on Time and Events for the group "0"
kmf.fit(T[ECG_data.group==1], ECG_data.alive[ECG_data.group==1], label='0')
kmf.plot(ax=ax)

#################################################### END #########################################

